components/sup/src/manager/service.rs

```diff
@@ -352,11 +356,13 @@ impl Service {
                    .join("hooks")
     }
 
+    #[allow(clippy::too_many_arguments)]
     pub async fn new(sys: Arc<Sys>,
                      spec: ServiceSpec,
                      manager_fs_cfg: Arc<FsCfg>,
                      organization: Option<&str>,
                      gateway_state: Arc<GatewayState>,
+                     butterfly: ButterflyServer,
```

Comment:
I know this is a draft and all, but further threading a Butterfly server down here squicks me out a little bit :grimacing: 

(I don't yet have anything more constructive than that, sorry :sweat_smile:)